### ISSUE TITLE
WatchFlower.exe just hangs with a blank screen

### ISSUE THREAD

**Issue by Stan-Gobien** [emericg/WatchFlower#47]
**Describe the bug**
The program worked fine for a couple of weeks.
Now since a couple of days it won't open anymore. The program starts to the Windows tray, but on right click "show" or "settings" a blank screen open with the title "Watchflower (not responding)" and just hangs. Waited a couple of minutes but no success. Taskmanager shows WatchFlower.exe with a constant 6% CPU (16T CPU, shows 1 thread almost at max constantly) and a steady 85.224KB RAM with no change. Seems like it's stuck in some kind of loop.

**Expected behavior**
I expected the program to open.
It worked (sort of) fine for a couple of weeks.

**To Reproduce**
Don't know how to reproduce on your system

**Screenshots**
![afbeelding](https://user-images.githubusercontent.com/44786091/133841819-4be609e2-8830-4a62-97b2-f19a121159bf.png)

![afbeelding](https://user-images.githubusercontent.com/44786091/133841904-bafdfe8b-978c-47eb-9ec6-3921a015bee1.png)

![afbeelding](https://user-images.githubusercontent.com/44786091/133841953-9dfa888c-c4c2-4fc1-b1c5-74bdccfd8225.png)

**Your environment**
Windows 10 x64 21H1
AMD Ryzen 7 3700X / 32GB RAM / SSD storage
Watchflower 2.1 x64 for windows zip from the site

**Additional context**
I tried renaming the "DB" file in %AppData%\WatchFlower and a new database is created but the problem is not solved.

**Comment by Stan-Gobien** [comment]
And ofcourse, after submitting the bug report I decide to check if WatchFlower saves settings in the Windows Registry.
Computer\HKEY_CURRENT_USER\SOFTWARE\WatchFlower
Deleted the Key, and sure enough the program starts again.

So problem solved (for now).

**Comment by Stan-Gobien** [comment]
Problem has reappeared now (the next day) when I start up my computer.

Deleted the reg key again, and works again.

**Comment by emericg** [comment]
Hi.
This is a bit weird given nothing has changed on WatchFlower side, but why not, with Windows you never know...

Can you try the newest version first: https://ci.appveyor.com/project/emericg/watchflower
There are some changes on the code that handle multiple launches (but maybe it's worse now :p)

Otherwise, to try and pinpoint what's wrong, can you try deleting only a couple of register keys from the WatchFlower folder?
All of the keys from ApplicationWindow/x,y, width, hieght, visibility... can have an impact on startup. Maybe settings/startMinimized and settings/trayEnabled). The others, very unlikely ...

**Comment by emericg** [comment]
A more direct link to download the latest version: https://ci.appveyor.com/project/emericg/watchflower/build/artifacts

**Comment by Stan-Gobien** [comment]
It was working today without issues.
Nevertheless, I updated to the latest build using your link.

**Comment by emericg** [comment]
Alright keep me posted if anything else happens.

**Comment by emericg** [comment]
Please re-open this issue if you have any new informations.

**Comment by Stan-Gobien** [comment]
Happens again.
![afbeelding](https://user-images.githubusercontent.com/44786091/144485608-c05ca642-7b8f-462e-a99b-93d6d228b7e6.png)

Watchflower remember the last position, on my secondary screen but hangs when I open it by double clicking on the tray icon (autostart).

**Comment by Stan-Gobien** [comment]
I looked at the registry key "HKEY_CURRENT_USER\SOFTWARE\WatchFlower\WatchFlower\ApplicationWindow"
The value for x was fffff9ee
The value for y was ffffffff

I set both values to 0 and now Watchflower opens fine on my primary screen.

I dragged the window back to the approximate same location on my secondary screen and closed the window.
The registry hex values for x and y are updated with new values (x = fffff66b; y = a7).
I closed the window and try to reopen by double click, it hangs again.
